File: backend/src/main/kotlin/com/example/backend/models/Magasin.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault

@Entity
@Table(name = "magasin")
class Magasin {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "code", nullable = true, length = 16)
  var code: String? = null

  @Column(name = "nom", nullable = false, length = 32)
  var nom: String? = null

  @ColumnDefault("0")
  @Column(name = "supprimer", nullable = false)
  var  supprimer: Int? = 0
}

File: backend_py/pharmaPython/app/models/magasin.py
from sqlalchemy import Column, Integer, String
from app.core.db import Base

class Magasin(Base):
  __tablename__ = "magasin"

  id = Column(Integer, primary_key=True)
  code = Column(String(16))
  nom = Column(String(32), nullable=False)
  supprimer = Column(Integer, default=0)
